pkg/installer/installer.go

```diff
@@ -113,137 +119,93 @@ func Install(ctx context.Context, args *Args) error {
 		}
 	}
 
-	print := func(s string) {
-		if !args.DryRun && !args.Quiet {
-			fmt.Print(s)
-		}
-	}
-
-	execute := func(name string, fn func(context.Context, *Args) error) error {
-		print(name + " ")
-
-		if err := fn(ctx, args); err != nil {
-			print(cross)
-			return err
+	execute := func(fn func(context.Context, *Args) error, totalSteps int) (err error) {
```

Comment:
We don't need this closure. Each component should accept `Args` , send start/progress/end messages. Because each component knows how many steps will be performed. They are fixed value.

`install` command knows how many components will be installed based on legacy flag. The progress at `install` command should initiate the value accordingly instead of having `TotalTasks` const.